sw/device/lib/dif/dif_aon_timer.c

```diff
@@ -4,7 +4,323 @@
+
+  // Watchdog control register should only be locked after the last
+  // control register access.
+  if (lock) {
+    aon_timer_watchdog_lock(aon);
+  }
+
+  return kDifAonTimerWatchdogOk;
+}
+
+dif_aon_timer_watchdog_result_t dif_aon_timer_watchdog_stop(
+    const dif_aon_timer_t *aon) {
+  if (aon == NULL) {
+    return kDifAonTimerWatchdogBadArg;
+  }
+
+  if (aon_watchdog_is_locked(aon)) {
+    return kDifAonTimerWatchdogLocked;
+  }
+
+  aon_toggle_watchdog_timer(aon, false);
+
+  return kDifAonTimerWatchdogOk;
+}
+
+dif_aon_timer_watchdog_result_t dif_aon_timer_watchdog_restart(
+    const dif_aon_timer_t *aon) {
+  if (aon == NULL) {
+    return kDifAonTimerWatchdogBadArg;
+  }
+
+  if (aon_watchdog_is_locked(aon)) {
+    return kDifAonTimerWatchdogLocked;
+  }
+
+  aon_clear_watchdog_counter(aon);
+  aon_toggle_watchdog_timer(aon, true);
+
+  return kDifAonTimerWatchdogOk;
+}
+
+dif_aon_timer_result_t dif_aon_timer_watchdog_get_count(
+    const dif_aon_timer_t *aon, uint32_t *count) {
+  if (aon == NULL || count == NULL) {
+    return kDifAonTimerBadArg;
+  }
+
+  *count = mmio_region_read32(aon->params.base_addr,
+                              AON_TIMER_WDOG_COUNT_REG_OFFSET);
+
+  return kDifAonTimerOk;
+}
+
+dif_aon_timer_result_t dif_aon_timer_watchdog_pet(const dif_aon_timer_t *aon) {
+  if (aon == NULL) {
+    return kDifAonTimerBadArg;
+  }
+
+  aon_clear_watchdog_counter(aon);
+
+  return kDifAonTimerOk;
+}
+
+dif_aon_timer_watchdog_result_t dif_aon_timer_watchdog_lock(
+    const dif_aon_timer_t *aon) {
+  if (aon == NULL) {
+    return kDifAonTimerWatchdogBadArg;
+  }
+
+  aon_timer_watchdog_lock(aon);
+
+  return kDifAonTimerWatchdogOk;
+}
+
+dif_aon_timer_result_t dif_aon_timer_watchdog_is_locked(
+    const dif_aon_timer_t *aon, bool *is_locked) {
+  if (aon == NULL || is_locked == NULL) {
+    return kDifAonTimerBadArg;
+  }
+
+  *is_locked = aon_watchdog_is_locked(aon);
+
+  return kDifAonTimerOk;
+}
+
+dif_aon_timer_result_t dif_aon_timer_irq_is_pending(const dif_aon_timer_t *aon,
+                                                    dif_aon_timer_irq_t irq,
+                                                    bool *is_pending) {
+  if (aon == NULL || is_pending == NULL) {
+    return kDifAonTimerBadArg;
+  }
+
+  uint32_t index = 0;
+  if (!aon_timer_get_irq_index(irq, &index)) {
+    return kDifAonTimerError;
+  }
+
+  uint32_t reg = mmio_region_read32(aon->params.base_addr,
+                                    AON_TIMER_INTR_STATE_REG_OFFSET);
+
+  *is_pending = bitfield_bit32_read(reg, index);
+
+  return kDifAonTimerOk;
+}
+
+dif_aon_timer_result_t dif_aon_timer_irq_acknowledge(const dif_aon_timer_t *aon,
+                                                     dif_aon_timer_irq_t irq) {
+  if (aon == NULL) {
+    return kDifAonTimerBadArg;
+  }
+
+  uint32_t index = 0;
+  if (!aon_timer_get_irq_index(irq, &index)) {
+    return kDifAonTimerError;
+  }
+
+  uint32_t reg = mmio_region_read32(aon->params.base_addr,
```

Comment:
This isn't quite right for a W1C register - you just need to set the bits you want. This read modify write will clear all the pending bits as well as the one you wanted to clear